lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1030,6 +1063,37 @@ SwiftLanguageRuntimeImpl::GetNumChildren(CompilerType type,
   if (auto *eti = llvm::dyn_cast_or_null<swift::reflection::EnumTypeInfo>(ti)) {
     return eti->getNumPayloadCases();
   }
+  // Objects.
+  if (auto *rti =
+      llvm::dyn_cast_or_null<swift::reflection::ReferenceTypeInfo>(ti)) {
+
+    switch (rti->getReferenceKind()) {
+    case swift::reflection::ReferenceKind::Weak:
+    case swift::reflection::ReferenceKind::Unowned:
+    case swift::reflection::ReferenceKind::Unmanaged:
+      // Weak references are implicitly Optionals, so report the one
+      // child of Optional here.
+      if (GetWeakReferent(*ts, type))
+        return 1;
+      break;
+    default:
+      break;
+    }
+
+    if (!tr)
+      return {};
+
+    auto *reflection_ctx = GetReflectionContext();
+    auto tc = swift::reflection::TypeConverter(reflection_ctx->getBuilder());
+    LLDBTypeInfoProvider tip(*this, *ts);
+    auto *cti = tc.getClassInstanceTypeInfo(tr, 0, &tip);
+    if (auto *rti =
+            llvm::dyn_cast_or_null<swift::reflection::RecordTypeInfo>(cti)) {
+      return rti->getNumFields();
```

Comment:
Should this be a recursive case or will there only ever be a record type underneath a class? (Well, probably).

Reply:
Currently it's only ever a RecordTypeInfo:

https://github.com/apple/swift/blob/4f3f68ea48286278b2c071a6ec1320cb220e1eea/stdlib/public/Reflection/TypeLowering.cpp#L2229-L2233

Reply:
We could factor out a helper function `GetNumFieldsFromTypeInfo()` that calls itself here, but the savings are marginal.

Reply:
I'll soon be implementing `GetNumFields`, and which may call for some refactoring.

I have a question, by "calls itself" do you mean recursive like in your top comment? If so, can you elaborate on what needs to be recursive?